Sources/HTTP/HTTPServer.swift

```diff
@@ -1,54 +1,336 @@
-// This source file is part of the Swift.org Server APIs open source project
 //
-// Copyright (c) 2017 Swift Server API project authors
-// Licensed under Apache License v2.0 with Runtime Library Exception
+//  HTTPServer.swift
+//  HTTP
 //
-// See http://swift.org/LICENSE.txt for license information
+//  Created by Helge Hess on 22.10.17.
+//  Copyright © 2017 ZeeZide GmbH. All rights reserved.
 //
 
-/// Definition of an HTTP server.
-public protocol HTTPServing: class {
+import Foundation
+import Dispatch
+#if os(Linux)
+  import Glibc
+#else
+  import Darwin
+#endif
 
-    /// Start the HTTP server on the given `port`, using `handler` to process incoming requests
-    func start(port: Int, handler: @escaping HTTPRequestHandler) throws
+public class HTTPServer {
+  
+  public enum SocketError : Swift.Error {
+    case setupFailed      (Int32)
+    case couldNotSetOption(Int32)
+    case bindFailed       (Int32)
+    case listenFailed     (Int32)
+  }
+  public enum Error : Swift.Error {
+    // TBD: this could be done in a better way
+    case socketError(SocketError)
+  }
+  
+  private  var boundAddress   : sockaddr_in? = nil
+  
+  private  var listenSource   : DispatchSourceRead?
+  
+  internal let queue =
+                 DispatchQueue(label: "de.zeezide.swift.server.http.server")
+  private  var connections = [ HTTPConnection ]()
+  
+  private  var acceptCount       = 0
+  private  let handlerBaseQueues : [ DispatchQueue ] = {
+    var queues = [ DispatchQueue ]()
```

Comment:
nit spacing `var queues = [DispatchQueue]()`

Reply:
Yeah, that is ugly. But feel free to run one of your style 'cleanups', I don't care.

Reply:
👍 Sounds good